<a href="https://colab.research.google.com/github/Mohdd-Afaan/data-science-master-2.0/blob/main/Statistics_Advance_4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would
use each type of test.

Ans: The key difference between a t-test and a z-test lies in the circumstances under which they are appropriate, primarily related to the sample size and whether the population standard deviation is known.
T-Test:
Applicability: Typically used when the sample size is small (usually less than 30) or when the population standard deviation is unknown.
Example Scenario: Suppose you have a small sample size (e.g., 20) and want to test if the mean of this sample is significantly different from a known population mean.
Z-Test:
Applicability: Typically used when the sample size is large (usually greater than 30) and the population standard deviation is known.
Example Scenario: Imagine you have a large sample size (e.g., 100) and you know the standard deviation of the entire population. You want to test if the mean of your sample significantly differs from the known population mean.

Q2: Differentiate between one-tailed and two-tailed tests.

Ans: One-Tailed Test:
Definition: In a one-tailed (or one-sided) test, the critical region or rejection region is located entirely in one tail of the distribution (either the left or the right).
Objective: Used when the hypothesis specifies the direction of the effect (e.g., greater than, less than).
Critical Region: Typically used to test if a parameter is significantly greater than or less than a hypothesized value.
Two-Tailed Test:
Definition: In a two-tailed (or two-sided) test, the critical region is divided between both tails of the distribution.
Objective: Used when the hypothesis does not specify the direction of the effect but only that there is a significant difference.
Critical Region: Typically used to test if a parameter is significantly different from a hypothesized value.

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for
each type of error.

Ans: Definition: Type 1 error occurs when the null hypothesis is incorrectly rejected when it is actually true. In other words, it is a false positive.
Example Scenario:
Hypotheses:
Null Hypothesis (H0): The drug has no effect on patients.
Alternative Hypothesis (H1): The drug has a positive effect on patients.
Type 1 Error: Concluding that the drug has a positive effect when, in reality, it does not.
Definition: Type 2 error occurs when the null hypothesis is not rejected when it is actually false. In other words, it is a false negative.
Example Scenario:
Hypotheses:
Null Hypothesis (H0): The new manufacturing process produces defective items at an acceptable rate.
Alternative Hypothesis (H1): The new manufacturing process produces defective items at an unacceptable rate.
Type 2 Error: Failing to detect that the new manufacturing process produces defective items at an unacceptable rate.

Q4: Explain Bayes's theorem with an example.

Ans: Bayes's Theorem is a mathematical formula that describes the probability of an event based on prior knowledge of conditions that might be related to the event.  The formula is expressed as:
P(A|B) = P(B|A).P(A) / P(B)
Where:
P(A|B) is the probability of event A occurring given that event B has occurred (the posterior probability).
P(B|A) is the probability of event B occurring given that event A has occurred (the likelihood).
P(A) is the prior probability of event A.
P(B) is the prior probability of event B.
Example:
A bag I contains 4 white and 6 black balls while another Bag II contains 4 white and 3 black balls. One ball is drawn at random from one of the bags, and it is found to be black. Find the probability that it was drawn from Bag I.
Solution:
Let's denote the events as follows:
Event A: Drawing a ball from Bag I.
Event B: Drawing a black ball.
We are asked to find the conditional probability P(A|B), which is the probability of drawing a ball from Bag I given that a black ball was drawn.
Using Bayes's Theorem:
P(B|A): Probability of drawing a black ball given that the ball is drawn from Bag I. Since Bag I contains 6 black balls out of 10, P(B|A) = 6/10
P(A) = P(B) : Probability of drawing a ball from Bag I. Since there are two bags and each is equally likely to be chosen, P(A) = 1/2
P(A|B) Probability of drawing a black ball given that the ball is drawn from Bag II. Since Bag II contains 3 black balls out of 7, P(A|B) = 3/7
P(B): Probability of drawing a black ball, considering both bags. This is given by the law of total probability: P(B) = P(A).P(B|A) + P(-A).P(B|-A)
where -A represents not drawing a ball from Bag I.
Therefore,
(6/10*1/2) / (6/10*1/2)+(3/7*1/2) = 7/12



Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.

Ans: A confidence interval (CI) is a statistical tool that provides a range of values within which we can be reasonably confident that the true value of a population parameter lies. It is often used to quantify the uncertainty associated with a sample estimate of a population parameter. The confidence level is a measure of the reliability of the interval, and it is often expressed as a percentage.
Example: Suppose we want to estimate the average height of a population of adults. We take a random sample of 30 individuals and find that the sample mean is 65 inches, and the sample standard deviation is 3 inches. We want to calculate a 95% confidence interval.
To calculate the 95% confidence interval for the population mean height based on the given sample, we can use the t-distribution since the population standard deviation is not known.
Since we have a sample size of 30, you have 29 degrees of freedom (df = n - 1).
We can use t-tables for this. For df = 29, the critical t-value is approximately 2.045.
Calculate the margin of error (ME):
ME= t.(s/root n)
ME= 2.045*(3/root 30) = 1.1200
Calculate the confidence interval:
Confidence Interval = mean +- ME = (65 + 1.1200= 66.12 , 65 - 1.1200= 63.88)
So, with 95% confidence, we estimate that the true average height of the population is between 63.88 inches and 66.12 inches.


Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the
event's probability and new evidence. Provide a sample problem and solution.

Ans: Bayes' Theorem is a fundamental principle in probability theory that allows us to update the probability of a hypothesis (or event) based on new evidence. The formula for Bayes' Theorem is:
 P(A|B) = P(B|A).P(A) / P(B) Where: P(A|B) is the probability of event A occurring given that event B has occurred (the posterior probability). P(B|A) is the probability of event B occurring given that event A has occurred (the likelihood). P(A) is the prior probability of event A. P(B) is the prior probability of event B.
 Example:
Suppose a factory produces light bulbs, and two machines, Machine 1 and Machine 2, are used to manufacture them. Machine 1 produces 60% of the bulbs, while Machine 2 produces 40%. The defective rate for bulbs produced by Machine 1 is 5%, and for Machine 2, it is 3%.
Now, a randomly selected bulb is found to be defective. What is the probability that it was produced by Machine 1?
Solution:
Let's define the events:
Event A: Bulb is produced by Machine 1.
Event B: Bulb is defective.
We want to find (A|B), the probability that the bulb is from Machine 1 given that it is defective.
P(A): Probability that a bulb is produced by Machine 1 = 0.60
P(B|A): Probability that a bulb is defective given it's from Machine 1 = 0.05
P(B): Probability that a bulb is defective given it's not from Machine 1 =
P(B) =   P(A).P(B|A) + P(-A).P(B|-A) = .60*.05 + .40*.03 = 0.042
Now
P(A|B) =  0.05 * 0.60 / 0.042 = 0.7142

Q7. Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation
of 5. Interpret the results.

Ans: To calculate the 95% confidence interval for a sample with a mean of 50 and a standard deviation of 5  we can use the t-distribution since the population standard deviation is not known. and assumming that sample size is 30.
Since we assummed a sample size of 30, you have 29 degrees of freedom (df = n - 1).
We can use t-tables for this. For df = 29, the critical t-value is approximately 2.045.
Calculate the margin of error (ME): ME= t.(s/root n) ME= 2.045*(5/root 30) = 1.8668
Calculate the confidence interval: Confidence Interval = mean +- ME = (50 + 1.8668= 51.8668 , 50 - 1.8668= 48.1332)
nterpretation:
With 95% confidence, we estimate that the true population mean is between 48.1332 and 51.8668. This means that if we were to take many random samples and calculate 95% confidence intervals for each, we would expect the true population mean to fall within these bounds in approximately 95% of the cases.

Q8. What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error.

Ans: The margin of error (MOE) in a confidence interval represents the range of values above and below the sample estimate within which the true population parameter is likely to lie. It is influenced by the level of confidence chosen and is calculated using the standard error of the sample mean.
As Sample Size Increases:
When the sample size (n) increases, the denominator of the formula (root n) increases.
A larger sample size reduces the standard error of the sample mean.
As a result, the margin of error decreases.
As Sample Size Decreases:
Conversely, when the sample size decreases, the standard error increases, leading to a larger margin of error.
Small Sample:
Sample Size : 20 students
Sample Mean : 65 inches
Sample Standard Deviation : 3 inches
Confidence Level: 95%
Degree of freedom: 20 -1 =19
Margin of Error (MOE) = 2.093(3/root 20) = 1.40
Large Sample:
Sample Size : 101 students
Sample Mean : 65 inches
Sample Standard Deviation : 3 inches
Confidence Level: 95%
Degree of freedom: 101 -1 = 100
Margin of Error (MOE) = 1.984(3/root 101) = 0.5922



Q9. Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population
standard deviation of 5. Interpret the results.

Ans: The z-score, also known as the standard score, measures how many standard deviations a particular data point is from the mean of a population. The formula for calculating the z-score is:
z = data point - population mean / standard deviation= 75 - 70 / 5= 1= .84134
Interpretation :
A z-score of 1 indicates that the data point (75) is higher than about 84.13% of the values in the population. In other words, approximately 84.13% of the data falls below a value of 75 in a normal distribution with a mean of 70 and a standard deviation of 5.
So, you can say that the data point of 75 is above the average and is in the upper 84.13% of the population, considering the given mean and standard deviation.

Q10. In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average
of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is
significantly effective at a 95% confidence level using a t-test.

In [2]:
import scipy.stats as stats

In [3]:
stats.t.ppf(1-0.05/2,49)

2.009575234489209

Ans: H0 = popilation mean = 0 (there is no effect)
    H1 = population mean != 0 (there is an effect of medication)
    sample_mean = 6
    significance_level = 0.05
    smaple_size = 50
    degree of freedom = 50 - 1 = 49reject H0.
    Calculate the t-statistic:
    t = 6 - 0 / (2.5/root 50) = 16.970725
    critical_value = 2.0095
    Conclusion:
    Since, 16.970 > 2.0095 we reject H0


Q11. In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95%
confidence interval for the true proportion of people who are satisfied with their job.

In [4]:
import math

In [7]:
#margin of error
math.sqrt(.65*(1-.65)/500)

0.02133072900770154

In [21]:
#z-value
stats.norm.ppf((1 + 0.95) / 2)

1.959963984540054

Ans: To calculate the 95% confidence interval for the true proportion of people satisfied with their job, we can use the following formula:
Confidence Interval = p +- z root p(1-p)/n
Where:
p : is the sample proportion (65% or 0.65 in decimal form),
z : is the critical z-value for a 95% confidence interval,
n : sample size 500
For a 95% confidence interval, the critical z-value is approximately 1.96
Confidence Interval= 0.65 + 1.96 * 0.021 = 0.654116 , 0.65 - 1.96 * 0.021 = 0.645884
Therefore, with 95% confidence, we estimate that the true proportion of people satisfied with their job is between 65.4116% and 64.5884%.

Q12. A researcher is testing the effectiveness of two different teaching methods on student performance.
Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82
with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a
significant difference in student performance using a t-test with a significance level of 0.01.

Ans: Formulate Hypotheses:
Null Hypothesis: No significant difference in mean scores between the two teaching methods
Alternative Hypothesis: There is a significant difference in mean scores between the two teaching methods
Significance level = 0.01
Calculate the Test Statistic:

In [16]:
t_stat, p_value = stats.ttest_ind(a=[85,6],b=[82,5])
print(f"t_stat:{t_stat},p_value:{p_value}")
alpha = 0.01
if p_value < alpha:
    print("Reject the null hypothesis. There is a significant difference.")
else:
    print("Fail to reject the null hypothesis. There is no significant difference.")

t_stat:0.036258906479596675,p_value:0.9743695041223197
Fail to reject the null hypothesis. There is no significant difference.


Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean
of 65. Calculate the 90% confidence interval for the true population mean.

In [28]:
stats.norm.ppf((1+0.90)/2).round(3)

1.645

In [33]:
8/math.sqrt(50)

1.131370849898476

Ans:  To calculate the 90% confidence interval we can use the following formula:
Confidence Interval = sample mean +- z * standard_deviation /root n
Confidence Interval = 65 + 1.645 * 1.131 = 66.860 , 65 - 1.645 * 1.131 = 63.139
Therefore, with 90% confidence, we estimate that the true proportion of people satisfied with their job is between 66.86 and 63.139.

Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average
reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to
determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

In [55]:
stats.t.ppf((1+0.90)/2,df=29).round(3)


1.699

In [54]:
from scipy.stats import t

# Given values
sample_mean = 0.25
sample_std_dev = 0.05
sample_size = 30
hypothesized_mean = 0

# Calculate the t-statistic
t_stat = (sample_mean - hypothesized_mean) / (sample_std_dev / (sample_size**0.5))

# Find critical t-values for a two-tailed test with 29 degrees of freedom (30 - 1) and alpha = 0.10
t_critical_left = t.ppf(0.05, df=29)
t_critical_right = -t_critical_left
print(t_critical_left)
# Compare the t-statistic with critical values
if t_stat < t_critical_left or t_stat > t_critical_right:
    print("Reject the null hypothesis. There is a significant effect of caffeine on reaction time.")
else:
    print("Fail to reject the null hypothesis. There is no significant effect of caffeine on reaction time.")


-1.6991270265334977
Reject the null hypothesis. There is a significant effect of caffeine on reaction time.


Ans: H0 = hypothizes mean = 0 (There is significant effect of caffine on reaction time)
H1 = hypothizes mean != 0 (There is no effect of caffine on reaction time)
sample_size = 30
sample_mean = 0.25
sample_std = 0.05
C.I = .90
significance_value = .10
calculate t-stats
0.25/ (0.05/ root 30) = 27.386
critical_t
+-1.699
decision_rule if t-stats < or > +- critical_t Then reject the null hypothesis. There is a significant effect of caffeine on reaction time else fail to reject the null hypothesis. There is no significant effect of caffeine on reaction time.
since 27.386 > 1.966 we reject the null hypothesis. There is a significant effect of caffeine on reaction time